lib/state/simperium/middleware.ts

```diff
@@ -60,6 +62,14 @@ export const middleware: S.Middleware = store => {
       case 'REVISIONS_TOGGLE':
         fetchRevisions(store, nextState);
         break;
+
+      case 'TOGGLE_ANALYTICS':
+        buckets.preferences.update('preferences-key', nextState.preferences);
```

Comment:
we will want to update `window.analyticsEnabled` here as well so that it occurs immediately, otherwise we introduce a window of time in which we send analytics even though someone has opted out